lib/Core/SQLiteBuildDB.cpp

```diff
@@ -126,12 +126,20 @@ class SQLiteBuildDB : public BuildDB {
       }
       if (result == SQLITE_OK) {
         result = sqlite3_exec(
+          db, ("CREATE TABLE keys ("
+               "id INTEGER PRIMARY KEY, "
+               "key STRING UNIQUE);"),
+          nullptr, nullptr, &cError);
```

Comment:
My knowledge of SQLite breaks down here... I hope, but am not 100% sure, that SQLite will realize that the index on keys.key is a covering index and it will never actually need the primary table. It feels like there should be a way to ensure that the data is only ever present in the index (the table itself is irrelevant), but I'm not sure if that is possible.

It might be worth verifying that SQLite isn't hitting the actual table here when doing key lookups, although I would need to play around with the query explainer to even know exactly what to look for.